lib/routes/api/sip-gateways.js

```diff
@@ -4,6 +4,26 @@ const {DbErrorBadRequest, DbErrorUnprocessableRequest} = require('../../utils/er
 const decorate = require('./decorate');
 const sysError = require('../error');
 
+const checkUserScope = async(req, voip_carrier_sid) => {
+  const {lookupCarrierBySid} = req.app.locals;
+  if (req.user.hasAccountAuth) {
+    const carrier = await lookupCarrierBySid(voip_carrier_sid);
+    if (!carrier) throw new DbErrorBadRequest('invalid voip_carrier_sid');
+    if (carrier.service_provider_sid !== req.user.service_provider_sid &&
+      (carrier.account_sid !== req.user.account_sid)) {
```

Comment:
@guilherme-rauen I have trouble here allowing carrier.account_sid === null 

this is the same in the smpp-gateways.js